geoportal/geoportailv3_geoportal/lib/lingua_extractor.py

```diff
@@ -0,0 +1,378 @@
+import os
+import re
+import traceback
+import sys
+import urllib
+from urllib.parse import urlencode
+import httplib2
+import json
+import yaml
+from geojson import loads as geojson_loads
+import sqlalchemy
+from sqlalchemy.exc import NoSuchTableError, OperationalError, ProgrammingError
+from lingua.extractors import Extractor, Message
+
+import c2cwsgiutils.db
+import c2cgeoportal_commons.models
+from c2cgeoportal_commons.config import config
+from c2cgeoportal_geoportal.lib.bashcolor import RED, colorize
+
+import logging
+log = logging.getLogger(__name__)
+
+
+class LuxembourgExtractor(Extractor):  # pragma: no cover
+    """
+    ESRI legend extractor
+    """
+    extensions = [".ini"]
+
+    def __init__(self):
+        super().__init__()
+        if os.path.exists("geoportal/config.yaml"):
+            config.init("geoportal/config.yaml")
+            self.config = config.get_config()
+        else:
+            self.config = None
+        if os.path.exists("project.yaml"):
+            with open("project.yaml") as f:
+                self.package = yaml.safe_load(f)
+        else:
+            self.package = None
+        self.env = None
+        self.messages = []
+        self.fields = set()
+        self.TIMEOUT = 15
+
+    def __call__(self, filename, options, fileobj=None, lineno=0):
+        print('Entering into %s extractor' % self.__class__)
+
+        del fileobj, lineno
+
+        try:
+            engine = sqlalchemy.engine_from_config(self.config, "sqlalchemy_slave.")
+            factory = sqlalchemy.orm.sessionmaker(bind=engine)
+            db_session = sqlalchemy.orm.scoped_session(factory)
+            c2cgeoportal_commons.models.DBSession = db_session
+            c2cgeoportal_commons.models.Base.metadata.bind = engine
+            for dbsession_name, dbsession_config in self.config.get("dbsessions", {}).items():  # pragma: nocover
+                c2cgeoportal_commons.models.DBSessions[dbsession_name] = c2cwsgiutils.db.create_session(
+                    config, dbsession_name, **dbsession_config
+                )
```

Comment:
I'm more in favor of decoupling responsibility. This class should take the ORM session and the config as argument to the class.

Reply:
yes, this was a bit of a mess -> cleanup